frontend/src/components/Login.js

```diff
@@ -0,0 +1,138 @@
+    dispatch(user.actions.setAccessToken({ accessToken: null }));
+    dispatch(user.actions.setStatusMessage({ statusMessage: error.toString() }))
+    ;
+  };
+
+// Want to know the difference between the throw error message in the fetch and the error message we define in the endpoint in server.js. Is the one in the endpoint only for the backend, and the throw the one we should show in the frontend?
+
+  /* Handle sign up:
+  1. prevents the page from refreshing after the form is submitted.
+  2. Then does the fetch passing the name and password the user has inputted via the body as this is what the endpoint requires to create a new user.
+  3. If the name and password aren't ok because they don't match what is required e.g the lenght is less than 5, then the throw is actioned.
+  4. Otherwise if the fetch is successful then the data from the endpoint in server.js is returned in the json respons e.g. userId, accessToken, name and statusMessage.
+  5. If not successful then the handleLoginFailed function is called and the redux store is updated with an access token that's null and the status message from the endpoint in server.js.
+  6. Finally the setName and setPassword is cleared of the users name and password to an empty string.
+  7. This is what happens also for the handleLogin function.
+  */
+  const handleSignup = (event) => {
+    event.preventDefault();
+
+    // send data to backend, for saving in DB
+    fetch(SIGNUP_URL, {
+      method: "POST",
+      body: JSON.stringify({ name, password }),
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error(
+            "Signup failed. Please enter a valid name and password."
+          );
+        }
+        return res.json();
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFailed(err))
+      .finally(() => {
+        setName("");
+        setPassword("");
+      });
+  };
+
+  // Handle log in
+  const handleLogin = (event) => {
+    event.preventDefault();
+
+    fetch(LOGIN_URL, {
+      method: "POST",
+      body: JSON.stringify({ name, password }),
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error("Login failed. Please try again."); // throw redirects us to .catch
+        }
+        return res.json();
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFailed(err))
+      .finally(() => {
+        setName("");
+        setPassword("");
+      });
+  };
+
+  // If the user is successful (i.e. the accessToken has been created and successfully stored in the redux store) when logging in and signing in then the UserProfile component will be rendered. This is done because we have written the UserProfile tag in App.js, otherwise we could write it here and not in App.js
+  if (accessToken) {
+    return <></>;
+  }

```

Comment:
You should use React Router because having two returns in one component is not a good practice in React 🤓